src/Sylius/Bundle/ApiBundle/ApiPlatform/ConfigMergeManager.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\ApiPlatform;
+
+/** @experimental */
+class ConfigMergeManager
+{
+    public function mergeConfigs(...$configs): array
+    {
+        $resultingConfig = [];
+
+        foreach ($configs as $config) {
+            foreach ($config as $newKey => $newValue) {
+                $unsetNewKey = false;
+                if (is_string($newKey) && 1 === preg_match('/^(.*[^ ]) +\\(unset\\)$/', $newKey, $matches)) {
+                    [, $newKey] = $matches;
+                    $unsetNewKey = true;
+                }
+
+                if ($unsetNewKey) {
+                    unset($resultingConfig[$newKey]);
+
+                    if (null === $newValue) {
+                        continue;
+                    }
+                }
```

Comment:
`unsetNewKey` is set and used only in one place - this statement can be simplified
```suggestion
                if (is_string($newKey) && 1 === preg_match('/^(.*[^ ]) +\\(unset\\)$/', $newKey, $matches)) {
                    [, $newKey] = $matches;
                    
                    unset($resultingConfig[$newKey]);

                    if (null === $newValue) {
                        continue;
                    }
                }
```

In addition, can we extract `is_string($newKey) && 1 === preg_match('/^(.*[^ ]) +\\(unset\\)$/', $newKey, $matches)` to private method? Perhaps, this could be even extracted to separate service as well

Reply:
moved to private method as it uses variable reference, this way its easier to understand